applications/ConstitutiveLawsApplication/custom_utilities/tangent_operator_calculator_utility.h

```diff
@@ -207,6 +207,42 @@ class TangentOperatorCalculatorUtility
                 // Finally we compute the components
                 CalculateComponentsToTangentTensorSecondOrder(auxiliar_tensor, strain_plus, strain_minus, stress_plus, stress_minus, i_component);
 
+                // Reset the values to the initial ones
+                noalias(r_perturbed_strain) = unperturbed_strain_vector_gp;
+                noalias(r_perturbed_integrated_stress) = unperturbed_stress_vector_gp;
+            }
+        } else if (ApproximationOrder == 4) {
```

Comment:
4th order?

Reply:
Nope, in the enum is second order but with another approach. It is computed as first order but taking into account the second derivatives... (really close to central differences)

Reply:
```suggestion
        } else if (ApproximationOrder == 4) { // Second order but with another approach. It is computed as first order but taking into account the second derivatives
```